airbyte-integrations/connectors/source-slack/source_slack/components/threads_partition_router.py

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+
+from dataclasses import dataclass
+from typing import Iterable
+
+import dpath.util
+from airbyte_cdk.models import AirbyteMessage, SyncMode, Type
+from airbyte_cdk.sources.declarative.partition_routers import SubstreamPartitionRouter
+from airbyte_cdk.sources.declarative.types import Record, StreamSlice
+
+
+@dataclass
+class ThreadsPartitionRouter(SubstreamPartitionRouter):
+    """Overwrite SubstreamPartitionRouter to be able to pass more than one value
+    from parent stream to stream_slices
+    """
+
+    def stream_slices(self) -> Iterable[StreamSlice]:
+        if not self.parent_stream_configs:
+            yield from []
+        else:
+            for parent_stream_config in self.parent_stream_configs:
```

Comment:
Can we break down this large method into the smaller logical pieces for the better readability?

Reply:
updated